test/sparse/sparsevector.jl

```diff
@@ -589,7 +589,9 @@ let x = spv_x1, x2 = x2 = spv_x2
 
     # multiplies
     xm = SparseVector(8, [2, 6], [5.0, -19.25])
-    @test exact_equal(x .* x, abs2(x))
+    let y=x # workaround for broadcast not preserving sparsity in general
```

Comment:
`x .* x` gets transformed into `broadcast(x -> x*x, x)`, which doesn't produce a sparse result.  This will be fixed once @Sacha0 takes the code that he implemented for `SparseMatrix` (which makes `broadcast` for one or two sparse-matrix arguments produce a sparse matrix) and transfers/extends it to `SparseVector`.